tracer/src/Datadog.Trace/Debugger/Upload/DiagnosticsUploadApi.cs

```diff
@@ -67,5 +57,24 @@ await multipartRequest
             Log.Warning<int, string>("Failed to upload diagnostics with status code {StatusCode} and message: {ResponseContent}", response.StatusCode, content);
             return false;
         }
+
+        private Task<IApiResponse> PostAsync(string uri, ArraySegment<byte> data)
+        {
+            if (uri.Contains("diagnostics"))
```

Comment:
This seems a bit broad, are you sure you don't want to test for a specific path here 🤔 Given this is the "legacy" behaviour (I believe?) it might be a better option?

(If so, obviously don't have to do in this PR, for speed purposes)

Reply:
I thinks it's a good idea, I will add a work item for this.